### PR TITLE
Use cargo git dependencies instead of gitmodules.

### LINKED ISSUE #50: Move twitch auth flow to a separate repo
We can reuse the code in the tmux widget. To avoid copypasting it, create a new repo (already created twtich_oauth2_auth_flow), move the code there and make it available as a Rust library, and import the library from our projects.

### PR DESCRIPTION
**Pull request by satu**
Fixes #50

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -19,10 +19,9 @@ surf = "2.2"
 tiny_http = "0.6"
 tokio = { version = "1", features = ["full"] }
 toml = "0.4.2"
-twitch_api2 = { version = "0.5.0-alpha", features = ["client", "helix", "surf_client"], path="twitch_api2/" }
+twitch_api2 = { version = "0.5.0-alpha", features = ["client", "helix", "surf_client", "twitch_oauth2"], git = "https://github.com/Emilgardis/twitch_api2/"  }
```
Comment (LordMZTE) [inline Cargo.toml]: `version` does nothing here. Instead a git tag/branch/commit should be specified

File: Cargo.toml [hunk 2]
```diff
@@ -19,10 +19,9 @@ surf = "2.2"
...
 twitch-irc = { version = "2.2.0", features = ["refreshing-token"] }
-twitch_oauth2 = { version = "0.5.0-alpha", features = ["surf_client"], path = "twitch_api2/twitch_oauth2/" }
-twitch_oauth2_auth_flow = { version = "0.0.1-alpha", path = "twitch_oauth2_auth_flow/" }
+twitch_oauth2_auth_flow = { version = "0.0.1-alpha", git = "https://github.com/stuck-overflow/twitch_oauth2_auth_flow", branch = "main" }
```
Comment (LordMZTE) [inline Cargo.toml]: `version` does nothing

**Review by LordMZTE — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (satu) [reply Cargo.toml]: done

File: Cargo.toml (hunk 2 above)
Comment (satu) [reply Cargo.toml]: done

**Review by LordMZTE — APPROVED** [review]

**Review by ed-fish — APPROVED** [review]
Ok - maybe explain gitmodules to me on thurs?

**Comment by satu** [comment]
> Ok - maybe explain gitmodules to me on thurs?

I will! Everybody seems to hate them though.

**Pull request merged**